jerry-core/mem/mem-heap.cpp

```diff
@@ -40,12 +40,14 @@
 #ifdef JERRY_VALGRIND
 # include "memcheck.h"
 
-# define VALGRIND_NOACCESS_STRUCT(s)    (void)VALGRIND_MAKE_MEM_NOACCESS((s), sizeof (*(s)))
-# define VALGRIND_UNDEFINED_STRUCT(s)   (void)VALGRIND_MAKE_MEM_UNDEFINED((s), sizeof (*(s)))
-# define VALGRIND_DEFINED_STRUCT(s)     (void)VALGRIND_MAKE_MEM_DEFINED((s), sizeof (*(s)))
-# define VALGRIND_NOACCESS_SPACE(p, s)  (void)VALGRIND_MAKE_MEM_NOACCESS((p), (s))
-# define VALGRIND_UNDEFINED_SPACE(p, s) (void)VALGRIND_MAKE_MEM_UNDEFINED((p), (s))
-# define VALGRIND_DEFINED_SPACE(p, s)   (void)VALGRIND_MAKE_MEM_DEFINED((p), (s))
+# define VALGRIND_NOACCESS_STRUCT(s)     (void)VALGRIND_MAKE_MEM_NOACCESS((s), sizeof (*(s)))
+# define VALGRIND_UNDEFINED_STRUCT(s)    (void)VALGRIND_MAKE_MEM_UNDEFINED((s), sizeof (*(s)))
```

Comment:
Thank you for pointing this out. I've missed that.

Anyway, without removing the `(void)` the code isn't compiled successfully with the changes.